File: binary_search.js
function binary_search(low, high, arr, target) {
    while (low <= high) {
        let mid = Math.floor(low + (high - low) / 2);
        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
             low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return "not in array";
}
let nums = [];
for (let i = 0; i < 100; i++) {
    nums.push(i);
}
let n = nums.length;
console.log(binary_search(0, n-1, nums, 54));
console.log(binary_search(0, n-1, nums, 12));
console.log(binary_search(0, n-1, nums, 78));
console.log(binary_search(0, n-1, nums, 101));

File: binary_search.cpp
#include <iostream>
#include <vector>
using namespace std;
int binary_search(vector<int> arr, int start, int end, int target) {
    while (start <= end) {
        int med = start + (end - start) / 2;
    if (arr[med] == target) {
        return med;
    } else if (arr[med] < target) {
        start = med + 1;
    } else {
        end = med - 1;
    }
}
return -1;
}
int main() {
    vector<int> arr;
    for (int i = 0; i < 100; i++) {
        arr.push_back(i);
    }
    int target = 38;
    int start = 0;
    int end = arr.size() - 1;
    int result = binary_search(arr, start, end, target);
    cout << result;
}